src/deploy.ts

```diff
@@ -86,51 +102,121 @@ export class HypERC20Deployer extends GasRouterDeployer<
     );
   }
 
+  protected async deployCollateral(
+    chain: ChainName,
+    config: CollateralConfig & HypERC20Config,
+  ): Promise<HypERC20Collateral> {
+    this.logger(`Deploying collateral router on ${chain}...`);
+    const erc20 = new ERC20__factory()
+      .attach(config.token)
+      .connect(this.multiProvider.getProvider(chain));
```

Comment:
Nit, could this be just `const erc20 = ERC20__factory.connect(config.token, this.multiProvider.getProvider(chain))`